simple-facebook-instant-articles.php

```diff
@@ -400,15 +400,27 @@ public function render_pull_quotes( \DOMDocument &$dom, \DOMXPath &$xpath ) {
 	 */
 	public function render_images( \DOMDocument &$dom, \DOMXPath &$xpath ) {
 
-		// Images - with parent that's not <figure>.
+		// Get all images that are not children of figure already.
 		foreach ( $xpath->query( '//img[not(parent::figure)]' ) as $node ) {
 
-			$figure = $dom->createElement( 'figure' );
+			$figure   = $dom->createElement( 'figure' );
+			$top_node = $node;
 
-			$node->parentNode->replaceChild( $figure, $node );
+			// If image node is not a direct child of the body, we need to move it there.
+			// Recurse up the tree looking for highest level parent/grandparent node.
+			while ( $top_node->parentNode && 'body' !== $top_node->parentNode->nodeName ) {
+				$top_node = $top_node->parentNode;
+			}
```

Comment:
Thing is - I don't want the body node. But rather the top node that is a direct child of body. So that I can insert the image after it and keep things in the right order.

In reality - `top_node` is probably the image node parent. But I don't want to assume this to be the case.